### DISCUSSION TITLE
Disable TimeMachine access while SEB is running

### DISCUSSION THREAD

**Discussion by hannaaase** [SafeExamBrowser/seb-mac#345] [General]
Hi, the exam date is just over this weekend, and I have found that one can see all files while in SafeExam - via TimeMachine! (By clicking the icon up in right corner while in the program)

Also this is possible while screen recording is turned off... Does the system still pick up on the seconds a student is "inside TimeMachine"? Or does one manually need to check that every single Mac computer has screen recording turned on? 

Happy for input on this as a small disaster if everyone that has a Mac using TimeMachine back-up has an unfair advantage.

**Comment by danschlet** [comment]
Thank you for reporting this security issue. I was able to reproduce it and have implemented a fix, which I will include in SEB 3.3.1 (the current release candidate version on GitHub isn't yet updated, but I will upload an updated version with the fix for this issue asap).

There are two possible workarounds which would work with the current SEB version as well:
- Disabling the macOS menu bar (disable setting Preferences / [User Interface Pane](https://safeexambrowser.org/macosx/mac_usermanual_en.html#AppearancePane) / Show menu bar or settings key showMenuBar = false). 
- Using the AAC macOS lockdown mode: (enable setting Preferences / [Security Pane](https://safeexambrowser.org/macosx/mac_usermanual_en.html#SecurityPane) / Prefer Assessment Mode (AAC))

On macOS 12, 13 and 14, WiFi settings can then be accessed in the Control Center using the keyboard shortcut fn - c.

I don't know what kind of screen recording your exam provider is using, but they should make sure it cannot be switched off during an exam or the exam doesn't start when that feature is off.
